contracts/HubController.sol

```diff
@@ -91,26 +132,34 @@ contract HubController is Named, Versioned, ContractStatus, Ownable {
         }
     }
 
-    function setContractAddress(
-        string calldata contractName,
-        address newContractAddress
-    ) external onlyOwnerOrMultiSigOwner {
-        hub.setContractAddress(contractName, newContractAddress);
-    }
-
-    function setAssetStorageAddress(
-        string calldata assetStorageName,
-        address assetStorageAddress
-    ) external onlyOwnerOrMultiSigOwner {
-        hub.setAssetStorageAddress(assetStorageName, assetStorageAddress);
+    function _setParameters(bytes[] calldata setParametersEncodedData) internal {
+        address parametersStorageAddress = hub.getContractAddress("ParametersStorage");
+        for (uint i; i < setParametersEncodedData.length; ) {
+            this.forwardCall(parametersStorageAddress, setParametersEncodedData[i]);
```

Comment:
How would this work? We if we need to call multiple functions setR0, setR1, setR2, will all calls be in encodedData and `target.call{value: 0}(data);` would execute all of them?

Reply:
Pretty much, yes. Array of encoded data is constructed and passed as argument to HubController's function
<img width="847" alt="image" src="https://user-images.githubusercontent.com/71610423/230595480-f220fade-a19c-4416-880e-735b189bb96e.png">